ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpers.java

```diff
@@ -203,5 +205,34 @@ public static Ds3ClientHelpers wrap(final Ds3Client client) {
      * @throws IOException
      */
     public abstract Iterable<Ds3Object> listObjectsForDirectory(final Path directory) throws IOException;
-}
 
+    /**
+     *
+     * @param objectsList
+     * @param prefix
+     * @return
+     */
+    public abstract Iterable<Ds3Object> addPrefixToDs3ObjectsList(final Iterable<Ds3Object> objectsList, final String prefix);
+
+    /**
+     *
+     * @param objectsList
+     * @param prefix
+     * @return
+     */
+    public abstract Iterable<Ds3Object> removePrefixFromDs3ObjectsList(final Iterable<Ds3Object> objectsList, final String prefix);
+
+    /**
+     * Strip prefix from the beginning of objectName.  If objectName does not start with prefix, return objectName unmodified.
+     * @param objectName
+     * @param prefix
+     * @return
+     */
+    public static String stripLeadingPath( final String objectName, final String prefix) {
+        String returnString = objectName;
```

Comment:
I might change this to be `if (objectName.startsWith(prefix)) { return objectName.substring(prefix.length()); } return objectName`.  It doesn't require allocating a new variable.